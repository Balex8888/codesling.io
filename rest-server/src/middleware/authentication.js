import jwt from 'jsonwebtoken';

import log from '../lib/log';

export const generateToken = ({ username, id }) => {
  const token = {};

  token.accessToken = jwt.sign({
    exp: Math.floor(Date.now() / 1000) + (60 * 60),
    username,
    id,
  }, process.env.TOKEN_SECRET);

  // token.refreshToken = jwt.sign({
  //   username,
  //   id,
  // }, process.env.TOKEN_SECRET);

  return token;
};

// export const refreshToken = (req, res, next) => {
//   try {
//     const token = jwt.decode(req.body.token, process.env.TOKEN_SECRET);
//     const newToken = generateToken(req.body.token);
//     next(newToken);
//   } catch (e) {
//     log('error refreshing token');
//     next(e);
//   }
// };

export const verifyUserWithJWT = (req, res, next) => {
  try {
    jwt.verify(req.headers.authorization.slice(7), process.env.TOKEN_SECRET);
    log('token verified');
    next();
  } catch (e) {
    log('token not verified');
    next(e);
  }
};

// export const logout = async (req, res, next) => {
//   const { accessToken } = req.body;
//   try {
//     // delete the token
//   } catch (e) {
//     log('error logging out');
//     next(e);
//   }
// };
