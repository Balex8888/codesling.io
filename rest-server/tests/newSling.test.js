import axios from 'axios';
import dotenv from 'dotenv';

import { generateToken } from '../src/middleware/authentication';

dotenv.config();
dotenv.load();

const URL = `${process.env.HOST}:${process.env.PORT}`;

test('rest-server should respond with a 200 to {GET /api/new-sling}', async () => {
  let resp = {};
  const { accessToken } = generateToken({username: 'test', id: 1 });
  try {
    resp = await axios.get(URL + '/api/new-sling', {
      headers: {
        Authorization: `Bearer ${accessToken}`,
      }
    });
  } catch (e) {
    console.log('Unexpected error for {GET /api/new-sling}. e = ', e);
  }
  expect(resp.status).toMatchSnapshot();
});
