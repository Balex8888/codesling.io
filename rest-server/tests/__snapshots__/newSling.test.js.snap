// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rest-server should respond with a 200 to {GET /api/new-sling} 1`] = `200`;
